lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -853,35 +895,7 @@ export function detectTestingLibraryUtils<
 
 		const findImportedUserEventSpecifier: () => TSESTree.Identifier | null =
 			() => {
-				if (!importedUserEventLibraryNode) {
-					return null;
-				}
-
-				if (isImportDeclaration(importedUserEventLibraryNode)) {
-					const userEventIdentifier =
-						importedUserEventLibraryNode.specifiers.find((specifier) =>
-							isImportDefaultSpecifier(specifier)
-						);
-
-					if (userEventIdentifier) {
-						return userEventIdentifier.local;
-					}
-				} else {
-					if (
-						!ASTUtils.isVariableDeclarator(importedUserEventLibraryNode.parent)
-					) {
-						return null;
-					}
-
-					const requireNode = importedUserEventLibraryNode.parent;
-					if (!ASTUtils.isIdentifier(requireNode.id)) {
-						return null;
-					}
-
-					return requireNode.id;
-				}
-
-				return null;
+				return getUserEventImportIdentifier(importedUserEventLibraryNode);
```

Comment:
Here you should call `getUserEventImportIdentifier` only if `importedUserEventLibraryNode` exists.